src/esp/sim/BatchPlayerImplementation.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Meta Platforms, Inc. and its affiliates.
```

Comment:
I assume this code was moved from `BatchReplayRenderer.cpp` without changes so I'm not looking closely at this file.